### PR TITLE
feat(Mastodon): add new presence

### PR DESCRIPTION
**Pull request by Hikki0710**
## Description 
I added support for mastodon instances (for the moment I’m just putting the most used ones)

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

immagine
immagine

immagine
immagine

immagine
immagine

immagine
immagine

</details>

### CODE REVIEW

File: websites/M/Mastodon/metadata.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		"linuxrocks.online",
+		"universeodon.com",
+		"indieweb.social",
+		"infosec.exchange",
+		"aus.social",
+		"social.network.europa.eu"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/B0HPMOQ.png",
+	"thumbnail": "https://i.imgur.com/e7gLIsD.jpeg",
```
Comment (theusaf) [inline websites/M/Mastodon/metadata.json:26]: This thumbnail is a bit low-res. Try to find a higher-resolution version or upscale it if possible.

File: websites/M/Mastodon/presence.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+// checks if user is logged from textarea
+function checkPositionAccountAvatar() {
+	if (document.querySelectorAll(".autosuggest-textarea__textarea").length < 1)
+		return 1;
+	return 2;
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/jlQv2if.png",
```
Comment (theusaf) [inline websites/M/Mastodon/presence.ts:15]: Assets should be 512x512px in size.
Also, as your presence uses many assets, put them in an enum like so:
https://github.com/PreMiD/Presences/blob/3aa7e3fb8e54999012de1d1528f37bc907ae4213/websites/C/Crunchyroll/presence.ts#L52-L58

File: websites/M/Mastodon/presence.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	};
+	
+	if (document.location.pathname === "/home") {
```
Comment (theusaf) [inline websites/M/Mastodon/presence.ts:18]: `document.location` is used many times in this presence. Destructure the properties to simplify the code a bit:
```ts
const { pathname } = document.location;
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by DarkVillager** [comment]
@Hikki0710 Also run prettier.

if you need help go to discord: https://discord.com/channels/493130730549805057/607524579874832446 or dm me on discord: Dark_Ville#2756

File: websites/M/Mastodon/presence.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+							break;
+						}
+						case "/privacy-policy": {
+							presenceData.smallImageText = "Privacy and Policy";
+							presenceData.details = "Reading Privacy and Policy";
+
+							break;
+						}
+						// No default
```
Comment (theusaf) [inline websites/M/Mastodon/presence.ts:115]: Remove automated eslint comment
```suggestion
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/Mastodon/presence.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
+const presence = new Presence({
+	clientId: "1107321415243415574",
+});
+
+enum Assets {
+	Logo = "https://i.imgur.com/njcb4Wp.png",
+	FediverseLogo = "https://i.imgur.com/B0HPMOQ.png",
+	Search = "https://i.imgur.com/WhzGXp1.png",
```
Comment (DarkVillager) [inline websites/M/Mastodon/presence.ts:8]: Why use a different Asset for this?

File: websites/M/Mastodon/presence.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
+const presence = new Presence({
+	clientId: "1107321415243415574",
+});
+
+enum Assets {
```
Comment (DarkVillager) [inline websites/M/Mastodon/presence.ts:5]:
```suggestion
const enum Assets {
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/Mastodon/presence.ts
Comment (theusaf) [inline websites/M/Mastodon/presence.ts:1]: This file is not formatted properly. Use `yarn format` to format it.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/Mastodon/presence.ts [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+					presenceData.smallImageText = "Settings";
+					presenceData.details = "Viewing their settings";
+				} else if (pathname === "/relationships") {
+					presenceData.smallImageText = "Relationships";
+					presenceData.details = "Viewing their followers";
+				} else if (pathname.startsWith("/tags")) {
+					presenceData.smallImageText = "tags";
+					presenceData.details = `Searching ${pathname.split("/")[2]} tag`;
+				} else {
+					switch (pathname) {
```
Comment (theusaf) [inline websites/M/Mastodon/presence.ts:86]: You have a lot of nested switch statements and if statements that could be flattened. 
Consider putting all the static pathnames in a single switch, and the more dynamic conditions in if/else in the default area. Alternatively, use a `switch(true)` kind of structure.

File: websites/M/Mastodon/presence.ts [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+						case "/about": {
+							presenceData.smallImageKey = document
+								.querySelectorAll(".account__avatar")
+								[checkPositionAccountAvatar()].querySelector("img")
+								.getAttribute("src");
+							presenceData.largeImageKey = document
+								.querySelector(".about__header__hero")
+								.querySelector("img")
```
Comment (theusaf) [inline websites/M/Mastodon/presence.ts:94]: These query statements can be combined
```suggestion
								.querySelector(".about__header__hero img")
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/Mastodon/presence.ts [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+			presenceData.smallImageText = "Fediverse";
+			presenceData.details = "Looking in fediverse";
+			break;
+		}
+		case "/bookmarks":
+		case "/favourites":
+		case "/lists": {
+			const pathName = pathname.split("/")[1];
+			presenceData.smallImageText = pathName;
+			presenceData.details = `Looking their ${pathName}`;
```
Comment (theusaf) [inline websites/M/Mastodon/presence.ts:50]:
```suggestion
			presenceData.details = `Looking at their ${pathName}`;
```

**Review by theusaf — APPROVED** [review]

File: websites/M/Mastodon/metadata.json [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+		"securitymastod.one",
+		"linuxrocks.online",
+		"universeodon.com",
+		"indieweb.social",
+		"infosec.exchange",
+		"aus.social",
+		"social.network.europa.eu"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/B0HPMOQ.png",
```
Comment (Bas950) [inline websites/M/Mastodon/metadata.json:25]:
```suggestion
	"logo": "https://i.imgur.com/njcb4Wp.png",
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**